cmd/joy/execute.go

```diff
@@ -0,0 +1,186 @@
+			}()
+
+			return execute(cfg.CatalogDir, script.Path, args)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&list, "list", "l", false, "List available scripts")
+	cmd.Flags().BoolVarP(&jsonOutput, "json", "j", false, "Output as JSON")
+
+	return cmd
+}
+
+func selectScript(scripts []Script) (Script, error) {
+	var scriptNames []string
+	for _, script := range scripts {
+		scriptNames = append(scriptNames, script.Name)
+	}
+	prompt := &survey.Select{
+		Message: "Select script to execute:",
+		Options: scriptNames,
+	}
+	var index int
+	if err := survey.AskOne(prompt, &index); err != nil {
+		return Script{}, fmt.Errorf("prompting for script selection: %w", err)
+	}
+	return scripts[index], nil
+}
+
+type Script struct {
+	Name string `json:"name"`
+	Path string `json:"path"`
+}
+
+func getScripts(catalogDir, searchPath string) ([]Script, error) {
+	scriptsMap := make(map[string]Script)
+
+	catalogBinDir := filepath.Join(catalogDir, "bin")
+	files, err := os.ReadDir(catalogBinDir)
+	if err != nil {
+		return nil, fmt.Errorf("listing files under catalog bin dir %q: %w", catalogBinDir, err)
+	}
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
+		name := file.Name()
+		scriptsMap[name] = Script{
+			Name: name,
+			Path: filepath.Join(catalogBinDir, file.Name()),
+		}
+	}
+
+	searchDirs := filepath.SplitList(searchPath)
+	for _, dir := range searchDirs {
+		files, err := os.ReadDir(dir)
+		if os.IsNotExist(err) {
+			continue
+		}
+		if err != nil {
+			return nil, fmt.Errorf("listing files under search dir %q: %w", dir, err)
+		}
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
+			name := file.Name()
+			if !strings.HasPrefix(name, "joy-") {
+				continue
+			}
+			nameWithoutPrefix := strings.TrimPrefix(name, "joy-")
+			scriptsMap[nameWithoutPrefix] = Script{
+				Name: nameWithoutPrefix,
+				Path: filepath.Join(dir, name),
+			}
+		}
+	}
+
+	var scripts []Script
+	for _, script := range scriptsMap {
+		scripts = append(scripts, script)
+	}
+	sort.Slice(scripts, func(i, j int) bool {
+		return scripts[i].Name < scripts[j].Name
+	})
+
+	return scripts, nil
+}
+
+func formatScripts(scripts []Script) string {
+	var names []string
+	for _, script := range scripts {
+		names = append(names, script.Name)
+	}
+	return strings.Join(names, "\n")
+}
+
+func formatScriptsAsJson(scripts []Script) (string, error) {
+	b, err := json.MarshalIndent(scripts, "", "  ")
+	if err != nil {
+		return "", fmt.Errorf("marshalling scripts to JSON: %w", err)
+	}
+	return string(b), nil
+}
+
+func getScript(scripts []Script, scriptName string) (Script, error) {
+	for _, script := range scripts {
+		if script.Name == scriptName {
+			return script, nil
+		}
+	}
+	return Script{}, fmt.Errorf("script %q not found", scriptName)
+}
+
+func execute(catalogDir, scriptPath string, args []string) error {
+	cmd := exec.Command(scriptPath, args...)
+	cmd.Env = append(os.Environ(), fmt.Sprintf("JOY_CATALOG_DIR=%s", catalogDir))
```

Comment:
That seems to be a good enough way to tell the script where we are running. I was debating if the working directory should be the catalog dir, but your approach is more flexible.

I'll update `k8s-apply-vpa-recommendations.sh` once this is merged.